lib/index.js

```diff
@@ -5,6 +5,7 @@ module.exports = {
     'alias-model-in-controller': require('./rules/alias-model-in-controller'),
     'avoid-leaking-state-in-ember-objects': require('./rules/avoid-leaking-state-in-ember-objects'),
     'avoid-using-needs-in-controllers': require('./rules/avoid-using-needs-in-controllers'),
+    'balanced-braces': require('./rules/balanced-braces'),
```

Comment:
Can we actually rename this rule to `no-invalid-dependent-keys`? Why?

1. This rule could catch any invalid syntax. Balanced braces is just one part, but this rule can also look for other invalid characters, invalid position of `@each`, `[]`, etc. Note that you don't have to implement everything now but we should name the rule so that it can be expanded afterward.
2. This name would follow the pattern of the existing rule name [no-duplicate-dependent-keys](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-duplicate-dependent-keys.md).